target/linux/ramips/files/drivers/net/ethernet/ralink/gsw_mt7620.c

```diff
@@ -202,8 +202,8 @@ static void mt7620_hw_init(struct mt7620_gsw *gsw, int mdio_mode)
 	mtk_switch_w32(gsw, 0x7f7f7fe0, 0x0010);
 
 	/* setup port 4 */
-	if (gsw->port4 == PORT4_EPHY) {
-		u32 val = rt_sysc_r32(SYSC_REG_CFG1);
```

Comment:
Is removing the declaration here intended?

Reply:
There's a mix of use of block local declarations of val and use of the procedure local declaration of val.  In every case they're `u32`.  I suppose it's a style choice, but perhaps they should all be the same...

Reply:
removed because it is useless to declare more than once in a function

Reply:
@mpratt14 -- uh, it's not "useless" -- it means something completely different:  a variable in a block local declaration exists only for the scope of the block in which it is declared, and it shadows any enclosing declarations by the same name -- they are distinct variables.   If you have something like
```
   u32 val = 42;
   {
      u32 val = 13;
      foo1(val);
   }
   foo2(val);
```
... the call to `foo2` will be with 42.  If you wrote `val = 13;` instead of `u32 val = 13;` then the call to `foo2` would be with 13.   Fortunately, the code you're modifying does not rely on the procedure scoped variable `val` maintaining a value across blocks where it is currently declared locally -- so in this case, there's no harm in removing the block local declaration and using the procedure local `val`.   So _I_ don't object to the code as written except in as much as there's the mix of styles in the one procedure.

Reply:
sorry I mean, it is useless specifically in this function to declare it more than once

hw init function is only run once in the driver and the process is linear (the value only needed until the variable is redefined for something else).

I'm still learning C, so thanks for that clarification 👍🏼